### PR TITLE
Added GetAverageBlockTime

### PR DESCRIPTION
**Pull request by aditya520**
Added a method called GetAverageBlockTime in ethmonitor. It will be a good addition and is needed in Relayer Finalizer.

### CODE REVIEW

File: ethmonitor/chain.go [hunk 1]
```diff
@@ -55,6 +59,15 @@ func (c *Chain) push(nextBlock *Block) error {
 		c.blocks = c.blocks[1:]
 	}
 
+	// Update average block time
+	if n > 0 {
+		if c.averageBlockTime == 0 {
+			c.averageBlockTime = float64(nextBlock.Time() - headBlock.Time())
+		} else {
+			c.averageBlockTime = (c.averageBlockTime + float64(nextBlock.Time()-headBlock.Time())) / 2
+		}
```
Comment (attente) [inline ethmonitor/chain.go:68]: I think you could just move this into the `if` statement above instead.

Comment (attente) [reply ethmonitor/chain.go:68]: Also, what happens if `c.blocks` is a sequence of block removals due to reorg? I think it might be an issue when updating the average block time.

Comment (attente) [reply ethmonitor/chain.go:68]: Something else to think about is maybe we should consider computing average block time alternatively as (head block time - starting block time) / (head block number - starting block number) rather than doing a running average.

Comment (attente) [reply ethmonitor/chain.go:68]: > Also, what happens if `c.blocks` is a sequence of block removals due to reorg? I think it might be an issue when updating the average block time.

Actually, I suppose if there's a reorg, there should still be more added blocks than removed blocks, so the last block in the chain should still be an addition. So this _should_ be OK.

Comment (attente) [reply ethmonitor/chain.go:68]: > Something else to think about is maybe we should consider computing average block time alternatively as (next block time - starting block time) / (next block number - starting block number) rather than doing a running average.

I think the running average is fine too actually. It means we end up giving more recent measurements higher weight.

**Review by attente — APPROVED** [review]

File: ethmonitor/chain.go (hunk 1 above)
Comment (aditya520) [reply ethmonitor/chain.go:68]: We can think about the running average, as it will be more accurate for our needs. 
Maybe we can have a flag that we can pass, if we need a running average, or calculate the average from only the last `n` blocks. 🤔

Comment (attente) [reply ethmonitor/chain.go]: I think your current implementation is OK here as-is. We can refine later if we see unusual behaviour.

File: ethmonitor/chain.go [hunk 2]
```diff
@@ -25,17 +26,20 @@ func (c *Chain) clear() {
...
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	var headBlock *Block
+
 	// New block validations
 	n := len(c.blocks)
 	if n > 0 {
-		headBlock := c.blocks[n-1]
+		headBlock = c.blocks[n-1]
```
Comment (attente) [inline ethmonitor/chain.go:42]: BTW can you undo these changes?
